src/module-elasticsuite-catalog-optimizer/Ui/Component/Optimizer/Listing/DataProvider.php

```diff
@@ -66,6 +66,15 @@ public function __construct(
         parent::__construct($name, $primaryFieldName, $requestFieldName, $meta, $data);
 
         $this->collection = $collectionFactory->create();
+        // Add config column in order to get boost weight value.
+        $this->collection->addFieldToSelect('config');
+
+        // Join search container table in order to show request type in grid.
+        $this->collection->join(
+            ['search_container' => 'smile_elasticsuite_optimizer_search_container'],
+            'search_container.optimizer_id = main_table.optimizer_id',
+            ['search_container']
+        );
```

Comment:
The best approach is to use `$addFieldStrategies` and `$addFilterStrategies` if you're joining an extra table, it's less instrusive.
It might probably apply to `config` too.

Anyway, locally (with the third party AB testing module) 

- the request type column is empty
- I get an error when applying the filter

PS: I haven't tested yet on a legacy/pure CE 2.3.4

Reply:
I updated with the `$addFieldStrategies` and the column seems ok and the filtering is working
There was indeed a problem with the explain module which was overriding the optimizer data provider with inherit from it

Reply:
(https://github.com/Elastic-Suite/magento2-elasticsuite-premium/pull/54)